services/webhook/src/webhook/routes/webhook.py

```diff
@@ -90,10 +97,38 @@ def process_payload(
             logging.warning(
                 f"Webhook revision for {dataset} is the same as the current revision in the db - skipping update."
             )
-            return
+            return None
+        revision = payload["repo"].get("headSha")
+        old_revision: Optional[str] = None
+        for updated_ref in payload.get("updatedRefs") or []:
+            ref = updated_ref.get("ref")
+            ref_new_sha = updated_ref.get("newSha")
+            ref_old_sha = updated_ref.get("oldSha")
+            if ref == "refs/heads/main" and isinstance(ref_new_sha, str) and isinstance(ref_old_sha, str):
+                if revision != ref_new_sha:
+                    logging.info(f"headSha {revision} is different from newSha {ref_new_sha}")
```

Comment:
also: should we take it into account if it occurs? (maybe it's not important, since we should expect the payload to be always correct)

Reply:
I'm switching this to warning and modifying the text a bit.

It shouldn't occur but it's nice to have a warning in case we ever need to debug something related to that